### PR TITLE
Implemented BulkBeacon Protobuf Definitions version 2

### PR DESCRIPTION
**Pull request by antoniovl**
This PR implements a new definition for the BulkBeacon gRPC endpoint that enables the ability to separate latency from availability for performance.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,51 @@
-BUIlD_BASE_DIR := build
-GEN_BASE_DIR := gen
+# Copyright 2020 NSONE, Inc.
```
Comment (jroyalty) [inline Makefile:1]: Nit: Update copyright to 2021

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
+	mkdir -p $(BUILD_BASE_DIR)
+	go build -o $(BUILD_BASE_DIR)/grpc_example_client_v2 cmd/example_client_v2/main.go
 
 .PHONY: clean
 clean:
-	rm -rf $(GEN_BASE_DIR) $(BUIlD_BASE_DIR)
+	rm -rf $(BUILD_BASE_DIR) $(PB_OUT)/v2 $(PB_OUT)/v1
+
+.PHONY: hash
+hash:
```
Comment (jroyalty) [inline Makefile:47]: Still required?

File: pkg/bulkbeacon/auth.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 NSONE, Inc.
```
Comment (jroyalty) [inline pkg/bulkbeacon/auth.go:1]: Nit: Update copyright to 2021

File: pkg/bulkbeacon/auth.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bulkbeacon
+
+import "context"
```
Comment (jroyalty) [inline pkg/bulkbeacon/auth.go:17]: `context` is provided by the core protobuf libs?

File: cmd/example_client_v2/main.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+			Payloads: []*pb.Payload{
+				{
+					Metric: &pb.Payload_Latency{
+						Latency: &pb.LatencyMetric{
+							Value: 50,
+						},
+					},
+				},
+				{
+					Metric: &pb.Payload_PerfScore{
```
Comment (jroyalty) [inline cmd/example_client_v2/main.go:60]: How about a better example here?  Two payloads?  One with Latency and static avail score and another with the opposite?

File: README.md [hunk 6]
```diff
@@ -3,27 +3,42 @@ Pulsar Real User Monitoring (RUM)
...
+supported.
+
+See comments in the `bulkbeacon.proto` definition file for how to structure messages.
+Depending on the version of the Bulk Beacon ingestion, you will have to look at its respective
+directory (mentioned below in the `Getting started with gRPC` section).
+
+### Versions
+
+We currently support 2 protocol buffers + gRPC versions for Bulk Beacons. Each one can be
+found on `proto/bulkbeacon/{version}/bulkbeacon.proto`, with `version` taking values
```
Comment (jroyalty) [inline README.md:22]: Typo: `found on` -> `found in`

File: README.md [hunk 7]
```diff
@@ -3,27 +3,42 @@ Pulsar Real User Monitoring (RUM)
...
-messages.
+supported.
+
+See comments in the `bulkbeacon.proto` definition file for how to structure messages.
+Depending on the version of the Bulk Beacon ingestion, you will have to look at its respective
+directory (mentioned below in the `Getting started with gRPC` section).
+
+### Versions
+
+We currently support 2 protocol buffers + gRPC versions for Bulk Beacons. Each one can be
```
Comment (jroyalty) [inline README.md:21]: Maybe: ..."two formats for gRPC and HTTP+JSON"?

**Review by jroyalty — CHANGES_REQUESTED** [review]

File: cmd/example_client_v2/main.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 NSONE, Inc.
```
[the remark this answers is not in the document]
Comment (antoniovl) [reply cmd/example_client_v2/main.go:1]: Both examples were updated.

File: README.md [hunk 9]
```diff
@@ -3,27 +3,42 @@ Pulsar Real User Monitoring (RUM)
...
-messages.
+supported.
+
+See comments in the `bulkbeacon_v1.proto` and `bulkbeacon_v2.proto` definition files
+for how to structure messages.
+
+### Versions
+
+We currently support 2 protocol buffers + gRPC versions for Bulk Beacons. Each one can be
+found on `proto/bulkbeacon/{version}/bulkbeacon_{version}.proto`, with version taking values
```
[the remark this answers is not in the document]
Comment (antoniovl) [reply README.md]: Thanks.

File: Makefile (hunk 2 above)
Comment (antoniovl) [reply Makefile:47]: Not needed, deleted.

File: pkg/bulkbeacon/auth.go (hunk 4 above)
Comment (antoniovl) [reply pkg/bulkbeacon/auth.go:17]: By the gRPC service. The struct implements `grpc.PerRPCCredentials` interface, and one method signature has a parameter of type context.

File: README.md (hunk 6 above)
Comment (antoniovl) [reply README.md:22]: fixed

File: README.md (hunk 7 above)
Comment (antoniovl) [reply README.md:21]: Makes sense. Changed.

File: cmd/example_client_v2/main.go (hunk 5 above)
Comment (antoniovl) [reply cmd/example_client_v2/main.go:62]: Added more samples.

**Review by jroyalty — APPROVED** [review]

**Pull request merged**